### DISCUSSION TITLE
Suggestion: Subgroups in xit

### DISCUSSION THREAD

**Discussion by jlb6907** [jotaen/xit#57] [File Format]
Is it possible to add subgroups with tabs in the xit specification ?
```
Group 1
[ ] Item
[ ] Item
[ ] Item 

    Subgroup 1.1
    [ ] Item 
    [ ] Item

        Subgroup 1.1.1
        [ ] Item 
        [ ] Item

        Subgroup 1.1.2
        [ ] Item
        [ ] Item

    Subgroup 1.2
    [ ] Item
    [ ] Item

Group 2
[ ] Item
[ ] Item
[ ] Item 
```

I have a lot of subprojects inside my projects, and it's easier to understand and to use subgroups than subtasks.

For now, I'm using [this vscode ](https://github.com/fabiospampinato/vscode-todo-plus) plugin, but I would prefer to use xit format.

![image](https://user-images.githubusercontent.com/4329562/233541859-6a71053c-ae59-4e19-bdd1-2dd18457f329.png)

**Comment by jotaen** [comment]
Thanks for writing up and sharing your idea!

Similar to https://github.com/jotaen/xit/discussions/2, this unfortunately will be tricky to implement probably, due to limitations in syntax highlighting engines of various text editors. So in theory, such a nesting could be added to the specification document, but I think it would be hard to impossible to implement full and correct syntax highlighting in editor plugins.

Note that group titles are optional, so as far as I see, your sample might not be too different from “pure” subtasks if you’d leave out the titles:

```
Group 1
[ ] Item 1
[ ] Item 2

    Subgroup 1.2
    [ ] Subitem 1
```

(with titles) vs. (without titles)

```
[ ] Item 1
[ ] Item 2
    [ ] Subitem 1
```
